### ISSUE TITLE
Nemo to show REAL filename for .desktop files

### ISSUE THREAD

**Issue by lucypol** [linuxmint/nemo#2456]
```
 * Nemo version (nemo --version)
Nemo 4.6.4

 * Is issue with desktop or windowed nemo?
Windowed

 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
Mint 20

 * 32 or 64 bit
64-bit
 ```

**Issue**
Not showing the real filename for .desktop files

**Steps to reproduce**
list files in /usr/share/applications/

**Expected behaviour**
Showing real filename

**Other information**
I would like an option to show the real filename and not this hide the real name like Windows
This is partly why we left Windows, so we can see the real information, not what is spoon fed to us by deception

**Comment by Jeremy7701** [comment]
Also a problem in nemo 4.6.4 on LMDE4
How long has this .desktop thing been a "feature" of some Linux distros?
Note that you can find .desktop files scattered in multiple directories:-
`/etc/alternatives/`
`/etc/xdg/`
`/etc/xdg/autostart/`
`/home/user/.config/autostart/`
`/home/user/.kde/share/apps/`
`/home/user/.local/share/applications/`
`/usr/lib/gnome-settings-daemon-3.0/gtk-modules/`
`/usr/lib/nvidia/current/`
`/usr/share/app-install/desktop/`
`/usr/share/applications/`   etc....
**even**
`/usr/share/gnome/applications/nemo.desktop`

**Comment by lucypol** [comment]
The problem is not the existence of .desktop files, it is the displaying of them, I want to see the actual filename, not the label from the contents inside it, that's windows deception shit right there.

**Comment by lucypol** [comment]
> Also a problem in nemo 4.6.4 on LMDE4
> How long has this .desktop thing been a "feature" of some Linux distros?
> Note that you can find .desktop files scattered in multiple directories:-
> `/etc/alternatives/`
> `/etc/xdg/`
> `/etc/xdg/autostart/`
> `/home/user/.config/autostart/`
> `/home/user/.kde/share/apps/`
> `/home/user/.local/share/applications/`
> `/usr/lib/gnome-settings-daemon-3.0/gtk-modules/`
> `/usr/lib/nvidia/current/`
> `/usr/share/app-install/desktop/`
> `/usr/share/applications/` etc....
> **even**
> `/usr/share/gnome/applications/nemo.desktop`

You missunderstand the issue, try reading it again

**Comment by Jeremy7701** [comment]
Apologies for reading `/usr/share/applications/` as the only directory at fault - I'd like an option in nemo that allows all .desktop files to be shown, regardless of which directory they are in.
One complication is that the **real** filename often bears little resemblance to the application name that gets launched.

**Comment by Jeremy7701** [comment]
Dear lucypol,
Thanks for the lovely off-topic rant about you hate Windows...[me too].

Whilst nemo doesn't show what you want, there is a limited workround:-
The last post in [https://askubuntu.com/questions/17220/can-nautilus-display-a-desktop-file-by-its-real-name](https://askubuntu.com/questions/17220/can-nautilus-display-a-desktop-file-by-its-real-name) indicates one possibility (and yes it does work in nemo):-

`sudo ln -s /usr /USR`
If we browse in nemo /usr/share/applications, the nemo display will be unchanged.
If we browse in nemo /USR/share/applications, nemo will display the real file-names of the .desktop files.

**Comment by AlexFolland** [comment]
This issue is still relevant.  I'm posting this because it is more than 1 year later, to clarify the status.

**Comment by MarkJeronimus** [comment]
I agree that this should be implemented as i got really confused when my text editor showed a different filename than the one I started editing. It's counterproductive, and I lost about 5 minutes reading this thread and not finding a solution.

Saying that the workaround is Nautilus is not only wrong but also rude because this is the Github is Nemo. Besides, Nemo was my workaround for much more serious Nautilus shortcomings, and since I started using it, I discovered new features and eventually like it even more than I ever liked Nautilus!

**Comment by Jeremy7701** [comment]
There's a long-running (but interesting) discussion on the security implications of desktop files at [https://unix.stackexchange.com/questions/373239/what-is-the-advantage-of-desktop-files-without-executable-bit-set](https://unix.stackexchange.com/questions/373239/what-is-the-advantage-of-desktop-files-without-executable-bit-set)

**Comment by bernd-wechner** [comment]
I'm with the crew that want Nemo to be a file manager and not a deception engineer. Show me the files in a directory please, not some disguise. Please.

**Comment by Jeremy7701** [comment]
When was the last time you needed to view or access a desktop file by name?
The answer for nearly everyone would be `never`.
So why allow a general purpose file manager - such as **nemo** provide easy access?

**Comment by AlexFolland** [comment]
On the other hand, why would a file manager try to hide something about a file?  It's okay for these `*.desktop` files to be hidden by things like launcher menus, launcher widgets, and desktop background icon systems, but a file manager must always display the files verbatim.  I have personally been annoyed by Nemo when trying to browse my `*.desktop` files and see their correct file names, only to have my file manager betray me and hide the names, with no option to prevent it from doing so.

**Comment by bernd-wechner** [comment]
> When was the last time you needed to view or access a desktop file by name?
> The answer for nearly everyone would be `never`.
> So why allow a general purpose file manager - such as **nemo** provide easy access?

Methinks you miss the point entirely. It is definitely NOT the job of the _file_ manager to doctor the view of _files_. 

How often: rarely, indeed. And when you do, suddenly look at one (like for example on a piece of portable software just unzipped which includes one, then suddenly, confusion - as Nemo oddly displays two - or more - 'files' with the same name). 

You have the burden of justification the wrong way around here methinks. It is disappointing to say the least to find a decent and laudable _file_ manager doctoring the displayed names of any _files_. A right royal bother to a great many people on earth was Windows decisive effort to look more Mac like a long time ago and by default hide file extensions on their explorer. Works for grandma, but frustrates a fairly large body of users, whose first move on getting a new PC or account on one after running explorer and thinking WTF? is to turn that off.

**Comment by Jeremy7701** [comment]
Perhaps a `nemo` setting for desktop files (off by default) would be acceptable - provided you could persuade the nemo team of its utility?
It could be buried with the other minor settings via gsettings.

**Comment by AlexFolland** [comment]
>provided you could persuade the nemo team of its utility

Multiple people have come here to figure out why files aren't represented properly by Nemo in a particular directory, and a ton of rationale has been given in this thread already.

>buried

Displaying files as they are in the system verbatim should be the default for the file manager.  If it is even not disabled by default, obviously the setting should be exposed in the settings GUI in a prominent place, since this anti-feature of hiding correct file names breaks the expectation of the job of a file manager.

**Comment by MarkJeronimus** [comment]
Obviously they just want to cater the 'general population' instead of us hardcore users. Just like Windows Explorer also doctors both files and directories (e.g. translating `c:\Users` to whatever display language you have configured).

I've never used Explorer, even before Microsoft started these shenanigans, and as long as I used Windows and had Total Commander, I used _that_. Note that I also don't even think about suggesting any Linux *Commander (e.g. Double Commander or  Krusader), because they're all bulky, slow, and/or are missing simple features like thumbnailing.

**Comment by AlexFolland** [comment]
To cater to the "general population", the correct file names should be shown.  The "general population" expects the file manager to display the files.  There is no reason to navigate to that directory specifically in a file manager if not looking for the files with their correct file names.

**Comment by Jeremy7701** [comment]
This argument is getting painfully repetitive.
Either the Mint/LMDE team will accept this as an important priority (initially raised as an issue back in 2020) or they won't.

**Comment by AlexFolland** [comment]
The duration that an issue can be reproduced does not affect the inconvenience the issue presents.  It's been almost 2 years since I first commented here that the ticket is still relevant, and the ticket is still just as relevant.  Please stop trying to convolute the thread in order to diminish the legitimacy of the need for a fix.  If you're bothered by the ticket, you can unsubscribe.

**Comment by leigh123linux** [comment]
>  If you're bothered by the ticket, you can unsubscribe.

I have unsubscribed due to lack of interest in the niche non-issue.

**Comment by Jeremy7701** [comment]
+1

**Comment by geajack** [comment]
> When was the last time you needed to view or access a desktop file by name?

*ALL* the time! Any time I want to add or update .desktop files, which I do quite often in order to set up new shortcuts for programs that don't have them, or modify an existing one, maybe to change the icon or to add a new "action" to it.

Maybe other distros/WMs are different, but on Ubuntu, if I'm looking at .desktop files in the file manager, it's because I want to modify them. I never launch .desktop files from my file manager, they're exclusively for using from the dock or the "applications overview".

In any case, I just don't see why this is even something we need to argue about. Can't it just be a setting? To me this seems like it should be as trivial as hiding or showing hidden files (i.e. files whose names start with a dot). Just let me hit Ctrl+Shift+H or something to flip between "desktopified" and raw view.

**Comment by realjjaveweb** [comment]
> When was the last time you needed to view or access a desktop file by name? The answer for nearly everyone would be `never`. So why allow a general purpose file manager - such as **nemo** provide easy access?

Let me bring some new light onto this: 
🏮**The core difference here is what is the relation between the `.desktop` file and the user usage**
- 👤 for a "common" user/usage `.desktop` file is like a "Desktop shortuct" from Windows - if you move it to other user-visible folder, you still expect the shortcut to be the App name, not it's real filename
- 👨‍💻 **however,** for someone who is actually _**editing**_ the files - this is an on-daily-basis need and the purpose of this thread - so @Jeremy7701 you may be right in the first point, but you're wrong here

💡 **Actually, I would expect >> 2 << things:**: 💡 
1. for `.desktop` files in **system** directories or in a **hidden** directory tree (like `~/.local/share/applications`) the full filename is _**automatically**_ displayed
2. there should be an option in Preferences that would force "always display original filename for `.desktop` files"

**Comment by realjjaveweb** [comment]
I just noticed one thing - if you chmod `+x` (make the .desktop files executable - that's when the App name is shown - if you don't - you get to see the original filename - though this is probably an accidental feature, since you don't need the .desktop files to be executable in the first place...

**Comment by Phantomwise** [comment]
An option to always show the real filenames would be so nice... It's really weird and inconvenient to need to use the CLI or a secondary file manager just to rename a `.desktop` file because you can't do it in your otherwise great file manager. I often forget nemo is weird with `.desktop` files and accidentally edit the `name` field of files instead of renaming the filenames.

> When was the last time you needed to view or access a desktop file by name? The answer for nearly everyone would be `never`. So why allow a general purpose file manager - such as **nemo** provide easy access?

As things are, you can't see the filenames of `.desktop` files with nemo, and you can't rename them because it edits the content of the `Name=` field inside the file instead. It makes nemo unusable as a file manager when you want to manage `.desktop` files, which is a problem for a file manager, no matter how few people are affected.

Adding an option to treat `.desktop` files as real files would fix that problem, and wouldn't change anything for people who don't need it if it's off by default.

Also `.desktop` files have a lot more uses than just app launchers. Not all `.desktop` files have `Type=Application`

Personally I use `.desktop` files with `Type=Link` like Windows shortcuts (I feel terrible for saying this but it was a very convenient feature and I do miss it). Shortcuts to directories, shortcuts to files, shortcuts to locations over SMB, shortcuts to websites, etc. But nemo seems to treat all `.desktop` the same, so I can't see their filename or rename them. 

I might be one of 7 people who uses `.desktop` files as shortcuts... But I think the issue lies with people not knowing they can be used this way and the absence of an easy GUI way to create them like on Windows, otherwise I think they'd see a lot more use, especially from people just switching from Windows. Shortcuts are very convenient.

**Comment by MaciejPolanski** [comment]
I spent a few hours grappling with this problem. I came across this discussion, and I agree with those who dislike it.

In my case, I use a separate Firefox profile for each Google product. Some are started by a separate `.desktop` file, while others are started directly from `about:profiles` in Firefox.

I have just migrated to Trixie and wanted to import the old `.desktop` file. However, there was a conflict for some reason — perhaps they weren't compatible. I'm not sure if I was angrier or more amused.

When the old file `youtube.desktop` (a disguised Firefox) was dragged, it was reported as having a conflicting filename with `Przeglądarka WWW Firefox`. However, `Przeglądarka WWW Firefox` is the Polish sandboxed name of `firefox-esr.desktop`, not a file name, and file names were not conflicting.

Then it forced me to change the file name and suggested the sandboxed name as the new one!
I ended up with the following files: `firefox-esr.desktop` and `Przeglądarka WWW Firefox_foobar`.

Creating copies to edit into new Firefoxes using Ctrl+C Ctrl+V resulted in identical files that could not be distinguished from each other — it was impossible to tell which was the original and which was the copy.

I think Nautilus, currently called 'GNOME Files', is better at the moment. It do not have such problems (yet?)

**Comment by tuxfanmatze** [comment]
I still see an urgent need for action here! Why does Nemo hide only the .desktop files? All others are displayed with their full file names. This should be the default setting in Nemo.
Furthermore, this is very confusing for beginners.

- If it is important, an optional switch can be added.
- This does not happen in the Thunar file manager, where the full name is always displayed.

**Comment by MarkJeronimus** [comment]
>     * If it is important, an optional switch can be added.
> 
>     * This does not happen in the Thunar file manager, where the full name is always displayed.

This also happens in Windows Explorer, but only when there's a special hidden desktop.ini file. The user can decide to delete this file to show filesystem names again.

**Comment by Jeremy7701** [comment]
Who are these people who need to see or even edit their .desktop files?

**Comment by gianma63** [comment]
> Who are these people who need to see or even edit their .desktop files?

Here's one! Someone who, because this issue has been ignored in Mint for over five years, preferred to switch to Debian...

**Comment by MarkJeronimus** [comment]
It's not just ignorant folks who don't know "how to use GNU/Linux" that want to look for desktop files.

Last time I looked at desktop files was to transfer Wine shortcuts from one PC to another and fix the icon. Before that I was looking to change some MimeType associations. Before that I was looking at where the hell the desktop file sits to find where the hell an application is installed (turned out it was a FlatPak application and apparently they don't have desktop files)

Basically, until there's a GUI editor for desktop files, people need to access the raw files. Renamed or not (because that's what you, [Jeremy7701](https://github.com/Jeremy7701), just claimed was not needed).

**Comment by Jeremy7701** [comment]
Just typed .desktop into a nemo search of / folder.
It displayed 4,651 desktop files.

If you want to find the **names** of the .desktop files which reference a given mimetype,
Then `gio mime mimetype` works
Example:-
```
gio mime image/png

Default application for “image/png”: xviewer.desktop
Registered applications:
	org.gnome.eog.desktop
	chromium-browser.desktop
	com.github.maoschanz.drawing.desktop
	display-im7.q16.desktop
	firefox.desktop
	gimp.desktop
	org.gnome.gThumb.desktop
	org.kde.kolourpaint.desktop
	pix.desktop
Recommended applications:
	org.gnome.eog.desktop
	chromium-browser.desktop
	com.github.maoschanz.drawing.desktop
	display-im7.q16.desktop
	firefox.desktop
	gimp.desktop
	org.gnome.gThumb.desktop
	org.kde.kolourpaint.desktop
	pix.desktop
```
A nemo search of (for example)  'org.gnome.eog.desktop' found a desktop file named '/usr/share/applications/Eye of GNOME'

**Comment by BujuArena** [comment]
Why would anyone want to use a workaround to be able to use the basic functionality of their file manager?

**Comment by MarkJeronimus** [comment]
> A nemo search of (for example) 'org.gnome.eog.desktop' found a desktop file named '/usr/share/applications/Eye of GNOME'

You hit the nail on the head by showing the mismatch between Nemo and terminal emulator usage (on which Mint, and every other distro in existence, is still heavily reliant despite best efforts). You just forgot one thing. It's easy to go from filename to display name but not the other way around. Try referencing "Eye of Gnome", as found in Nemo, in a terminal. Tab-expansion doesn't work, neither does `ls | grep`.

**Comment by Jeremy7701** [comment]
Despite what nemo shows, no file called "Eye of GNOME" actually exists.
If you edit nemo's "Eye of GNOME", you will find that you are trying to edit "org.gnome.eog.desktop"
So this seems an excellent reason to hide all .desktop files from view!

**Comment by BujuArena** [comment]
That seems more like an excellent reason for displaying the real file name, because the wrong file name was shown.  Your logic is completely backward.

**Comment by MarkJeronimus** [comment]
> So this seems an excellent reason to hide all .desktop files from view!

That last statement makes no sense at all. As an analogy, it's like saying possessing a flashlight is an excellent reason not to buy any lights for your house.

**Comment by tuxfanmatze** [comment]
I hope a Nemo developer can make a statement here.

**Comment by Phantomwise** [comment]
> So this seems an excellent reason to hide all .desktop files from view!

There are excellent reasons both for hiding them and for showing them, that's not the issue.

If the developers made a conscious choice to display the name of applications instead of filenames for .desktop files, it's a perfectly valid choice, even if I personally disagree with it. But that doesn't preclude from having an option to show the actual filenames, which many people would find useful and have requested. It wouldn't change the default behavior so wouldn't negatively affect people who prefer seeing application names instead of filenames.

**Comment by Tuupertunut** [comment]
> Who are these people who need to see or even edit their .desktop files?

To add to the experiences: I do it all the time. Just a few weeks ago I was hand writing new .desktop files for some applications that only existed as AppImages and needed to copy them into my `$HOME/.local/share/applications` to display them in the app menu. Today I was teaching a colleague how the Linux application launching works and how it compares to Windows and MacOS, and I needed to show that the icons in the app menu are actually these .desktop files in a directory. Just now when I went to look at my .desktop files I noticed that a hack I had made a month ago had created a broken .desktop file and I removed it.

For all the same reasons as the people above, I would like to have their real filenames visible in Nemo.
